### PR TITLE
Good practices/front end error handling

### PR DESCRIPTION
**Pull request by hazzo**
Resolves first task of #4

### CODE REVIEW

**Comment by jherencia** [comment]
@hazzo Great work, I really like the approach.

I do not have big concerns so I do not want to stop the process. I have doubts of Internal+External vs Known+Unknown, but I do not have a clear position and I think this should not stop anything.

Go ahead! 💪

File: good_practices/front_end_error_handling/README.md [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+      uiProvider.showSnackBar(ErrorCodes.GENERIC_ERROR);
+    }
+
+    if (rethrowError) throw e;
+  }
+}
+```
+
+Also, we could benefit from UI frameworks implementations to capture and display these errors
+but this methods will help us better to handle **uncontrolled errors**. With these implementations
```
Comment (jherencia) [inline good_practices/front_end_error_handling/README.md:134]: typo: these

File: good_practices/front_end_error_handling/README.md [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+that we might not know, partially ignore or that are not correctly documented. So the main
+idea is to try to capture every possible known errors and transform them to our app error
+object. For those that fall outside our knowledge a generic error should be created to display
+a friendly message to the user and log all details to our error monitoring system to track what it's
+really happening in those specific scenarios.
+
+### Network errors
+
+Network errors fall under the category of external errors and are the more common ones.
+Usually this errors are omitted, and they are partially caught by the language while
```
Comment (jherencia) [inline good_practices/front_end_error_handling/README.md:156]: typo: these

File: good_practices/front_end_error_handling/README.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+they bubble up and break some ui implementation.
+
+On an ideal scenario, this errors should be consciously caught and mapped to our
```
Comment (jherencia) [inline good_practices/front_end_error_handling/README.md:159]: typo: these

**Review by jherencia — APPROVED** [review]

**Comment by hazzo** [comment]
> @hazzo Great work, I really like the approach.
> 
> I do not have big concerns so I do not want to stop the process. I have doubts of Internal+External vs Known+Unknown, but I do not have a clear position and I think this should not stop anything.
> 
> Go ahead! 💪

Great! So I'll merge and follow with the next task!

**Pull request merged**